core/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java

```diff
@@ -102,7 +110,7 @@ public String chooseEngineServerAlias(String keyType, Principal[] issuers,
   public void updateIdentityCredentials(PrivateKey key, X509Certificate[] certs)
       throws CertificateException {
     // TODO(ZhenLian): explore possibilities to do a crypto check here.
-    this.keyInfo = new KeyInfo(key, certs);
+    this.keyInfo = new KeyInfo(checkNotNull(key), checkNotNull(certs));
```

Comment:
Pass the name of the argument as the second argument to checkNotNull. Otherwise it won't be clear which value was null, which can make debugging production problems more difficult. `checkNotNull(key, "key")`. No need for a more complex string.